src/main/java/com/birdie/kafka/connect/smt/Outbox.java

```diff
@@ -0,0 +1,240 @@
+package com.birdie.kafka.connect.smt;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.header.Headers;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.apache.kafka.connect.transforms.util.SimpleConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Outbox implements Transformation<SourceRecord> {
+    public static final String DELETED_FIELD = "__deleted";
+    public static final String PARTITION_KEY_FIELD = "partition_key";
+    public static final String HEADERS = "headers";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Outbox.class);
+
+    private ObjectMapper objectMapper = new ObjectMapper();
+    private interface ConfigName {
+        String TOPIC = "topic";
+        String PARTITION_SETTING = "partition-setting";
+        String NUMBER_OF_PARTITION_IN_TOPIC = "num-partitions";
+        String DEBUG_LOG_LEVEL = "debug-log-level";
+    }
+
+    public static final ConfigDef CONFIG_DEF = new ConfigDef()
+        .define(Outbox.ConfigName.TOPIC, ConfigDef.Type.STRING, ConfigDef.Importance.MEDIUM, "The name of the topic to send messages to.")
+        .define(Outbox.ConfigName.PARTITION_SETTING, ConfigDef.Type.STRING, "partition-number", ConfigDef.Importance.MEDIUM, "Set to \"partition-number\" (default) to use the record's partition_number value, or set to \"partition-key\" to generate partition number using record's partition_key value")
```

Comment:
change `auto-partitioning` to `partition-setting` with 2 possible values so the difference between the two options is a bit clearer (imho)

Reply:
Way better 👍